js/classes/zone.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * A Zone is a small portion of the Game World. 
+ * The Game Engine will load and deal with one Zone at a time.
+ * A Dimension (once an actual class, now simply an in-game concept) will contain several Zones.
+ * Zones should not be created willy-nilly - the only way you should actually construct a Zone is from the static getZones() method!
+ * 
+ * A Zone will define the minimum (upper left) and maximum (bottom right) point which is going to be visible to the Camera -
+ * anything outside this range will never be visible.
+ * A Zone will also contain a load method: this ensures that all assets are downloaded, all entities are placed (INCLUDING CHAD), etc.
+ * 
+ * @author Devin Peevy
+ */
+class Zone {
+
+    /**
+     * This constructor is PRIVATE should NOT be used outside the Zone class. To load a Zone, use the Zone.getZones() static method.
+     * This constructor can be used INSIDE THAT METHOD to dictate the min/max point and loadFunction.
+     * @param {Vector} minBlock The FIRST (top left) block which should be visible in this Zone. (NOT PIXEL COORDINATES)
+     * @param {Vector} maxBlock The LAST (bottom right) block which should be visible in this Zone. (NOT PIXEL COORDINATES)
+     * @param {function} loadFunction A function which is going to download all necessary assets and add all necessary entities into the zone.
+     */
+    constructor(minBlock, maxBlock, loadFunction) {
+        this.minBlock = minBlock;
+        this.maxBlock = maxBlock;
+        this.load = () => {
+            // Clear all entities from previous Zone.
+            GAME.clearEntities();
+            // Clear all images/music from previous Zone.
+            ASSET_MGR.clearCache();
+            loadFunction();
+        };
+    };
+
+    // DYNAMIC GETTERS:
+
+    /**
+     * @returns {Vector} The Min (upper left) block that is visible in this Zone.
+     */
+    get MIN_BLOCK() {
+        return this.minBlock;
+    };
+
+    /**
+     * @returns {Vector} The Max (bottom right) block that is visible in this Zone.
+     */
+    get MAX_BLOCK() {
+        return this.maxBlock;
+    };
+
+    /**
+     * @returns {Vector} The minimum (x, y) coordinate visible in this Zone.
+     */
+    get MIN_PT() {
+        return Vector.multiply(this.MIN_BLOCK, Block.SCALED_SIZE);
+    };
+
+    /**
+     * @returns {Vector} The maximum (x, y) coordinate visible in this Zone.
+     */
+    get MAX_PT() {
+        // I want to actually be able to SEE the final block (not just touch its top left corner). 
+        // Therefore:
+        const ones = new Vector(1, 1);
+        return Vector.multiply(Vector.add(this.MAX_BLOCK, ones), Block.SCALED_SIZE);
+    };
+
+    // I am not sure if we are actually going to need the following two getters, but I include them just in case:
+
+    /**
+     * @returns {Vector} The size of the zone, in BLOCKS.
+     */
+    get BLOCK_SIZE() {
+        // We gotta include an extra block to account for 0.
```

Comment:
What is 0? Is that the last block? I don't understand why we need ones. @B1GD3VD0G

Reply:
Same concept as zero indexing: we actually want to include both the min and the max. Therefore, size is actually one greater than max-min.